cmd/argocd/commands/app.go

```diff
@@ -1674,8 +1759,17 @@ func waitOnApplicationStatus(acdClient apiclient.Client, appName string, timeout
 		}
 		_ = w.Flush()
 	}
-	printFinalStatus(app)
-	return nil, fmt.Errorf("Timed out (%ds) waiting for app %q match desired state", timeout, appName)
+	select {
+	case complete := <-delete:
+		if complete {
+			return nil, nil
+		} else {
+			return nil, fmt.Errorf("Timed out (%ds) waiting for app %q to be deleted", opts.timeout, opts.appName)
+		}
+	default:
+		printFinalStatus(app)
+		return nil, fmt.Errorf("Timed out (%ds) waiting for app %q match desired state", opts.timeout, opts.appName)
+	}
```

Comment:
I don't think this change is necessary (see main comment). All of our successful returns should be inside the `appEventCh` loop.